test/result.test.ts

```diff
@@ -204,4 +204,4 @@ test('safeUnwrap', () => {
         // @ts-expect-error
         result.safeUnwrap();
     }
-})
+});
```

Comment:
I'd like to get a few tests in here to verify the types work not only for `Err` and `Ok`, but when working with combined `Result`s.  I'll be happy to add the tests myself if you need.

Reply:
Cool. Let me know if I can help!

Also, related to the previous comment, I was able to get the correct intelissense of a mapped value after a flatMap for both mapErr and map, so I dare to believe it should work for the most common use cases at least but you can never be too sure so yeah, tests will definitely be helpful.